src/components/contactForm/ContactForm.js

```diff
@@ -1,32 +1,60 @@
 import React from "react";
 
 export const ContactForm = ({
-  contacts,
   name,
   setName,
   phone,
   setPhone,
   email,
   setEmail,
-  handleSubmit
+  handleSubmit,
 }) => {
   return (
+    // Some name have hyphens, some have apostrophes. Let’s add support for these characters now.
     <form onSubmit={handleSubmit}>
       {/* name input */}
-    <input type="text" placeholder="name" value={name}
-        onChange={({target}) => {setName(target.value)}}
-        required />
+      <input
+        className="checkname"
+        type="text"
+        placeholder="name"
+        value={name}
+        pattern="[a-zA-Z]+(?:(?:\. |[' ])[a-zA-Z]+)*"
+        onChange={({ target }) => {
+          setName(target.value);
+        }}
+        autocomplete="on"
+        required
+      />
       {/* phone numer input */}
-      <input type="number" placeholder='phone number'
-       pattern="[1-9][0-9]{2}-[1-9][0-9]{2}-[0-9]{4}" value={phone} onChange={({ target }) => { setPhone(target.value) }} required />
-      
-      {/* ^((\+\d{1,3}(-| )?\(?\d\)?(-| )?\d{1,3})|(\(?\d{2,3}\)?))(-| )?(\d{3,4})(-| )?(\d{4})(( x| ext)\d{1,5}){0,1}$ */}
+      {/* This regex pattern is for Autralia and lineline and mobile numbers. */}
+      <input
+        type="number"
+        placeholder="phone number"
+        pattern="/^\+(?:[0-9] ?){6,14}[0-9]$/"
+        value={phone}
+         onChange={({ target }) => {
+          setPhone(target.value);
+        }}
+        required
+      />
       {/* email input  */}
+      <input
+        className="checkemail"
+        type="text"
+        value={email}
+        // pattern=""
+        onChange={({ target }) => {
+          setEmail(target.value);
+        }}
+        placeholder="example@domain.com"
+        autocapitalize="off"
+        spellcheck="false"
+        inputmode="email"

```

Comment:
Error in console - should this be inputMode?

Reply:
fixed